index.bs

```diff
@@ -282,30 +266,164 @@ Loader instances are initially created with the internal slots described in the
+1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "%RegistryPrototype%", «[[Entries]]» ).
+1. ReturnIfAbrupt(_O_).
+1. Set _O_’s [[Entries]] internal slot to a new empty List.
+1. Return _O_.
+</emu-alg>
+
+<h3 id="properties-of-the-registry-constructor">Properties of the Registry Constructor</h3>
+
+The value of the \[[Prototype]] internal slot of the Registry constructor is the intrinsic object %FunctionPrototype%.
+
+Besides the internal slots and the length property (whose value is 0), the Registry constructor has the following properties:
+
+<h4 id="registry-prototype">Registry.prototype</h4>
+
+The value of Registry.prototype is %RegistryPrototype%.
+
+This property has the attributes { \[[Writable]]: false, \[[Enumerable]]: false, \[[Configurable]]: false }.
+
+<h3 id="registry-prototype-object">Properties of the Registry Prototype Object</h3>
+
+<h4 id="registry-prototype-constructor">Registry.prototype.constructor</h4>
+
+The initial value of Registry.prototype.constructor is %Registry%.
+
+<h4 id="registry-prototype-@@iterator">Registry.prototype[ @@iterator ]()</h4>
+
+When the @@iterator method is called it returns an Iterator object (ES2015 25.1.1.2) that iterates over the registry entries of the \[[Entries]] internal slot, returning each entry as a key value pair. The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Let _entries_ be a new List.
+1. For each _pair_ in _registry_.[[Entries]], do:
+  1. Let _key_ be _pair_.[[Key]].
+  1. Let _value_ be _pair_.[[Value]].
+  1. Append { [[Key]]: _key_, [[Value]]: _value_ } to _entries_.
+1. Return CreateListIterator(_entries_).
+</emu-alg>
+
+The value of the name property of this function is "[Symbol.iterator]".
+
+<h4 id="registry-prototype-get">Registry.prototype.get(key)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
+1. If _pair_ exists, then return _pair_.[[Value]]. Otherwise return *undefined*.
+</emu-alg>
+
+<h4 id="registry-prototype-set">Registry.prototype.set(key, entry)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Assert: _entry_ is an initialized RegistryEntry instance.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
```

Comment:
It's tempting, but I really think we want the registry to be a type-restricted version of Map: the keys can only be strings, and the values can only be RegistryEntry objects. It'd be nice to reuse some of the methods of `Map.prototype` but if we do that then someone could subvert the type invariant by doing e.g. `Map.prototype.set.call(registry, 666, "zomg")`.

So maybe the closest we can get to reuse is to have an internal \[\[RegistryMap]] property that contains a Map, and have all the %RegistryPrototype% methods do the type checks and then delegate to the map's operations. But it can't just call methods directly on that map, because `Map.prototype` could've been tampered with. So I think the operations would have to call out to "the original meaning of `Map.prototype.XXX`" etc etc.